python/cuda_parallel/tests/test_reduce.py

```diff
@@ -170,7 +169,7 @@ def test_device_sum_cache_modified_input_it(
     dtype_inp = numpy.dtype(supported_value_type)
     dtype_out = dtype_inp
     input_devarr = numba.cuda.to_device(numpy.array(l_varr, dtype=dtype_inp))
-    i_input = iterators.CacheModifiedInputIterator(input_devarr, modifier="stream")
+    i_input = cudax.CacheModifiedInputIterator(input_devarr, modifier="stream")
```

Comment:
This is the only part of this PR I'm feeling a bit uneasy about. I think it's very useful not combine namespaces ("honking great idea") unnecessarily.

`cudax.iterators` is clearer and avoids other issues in the long run (unnecessary imports; dependency creep).

Reply:
For now, I've removed iterators and algorithms from the top-level `cudax` namespace, but longer term, we should think about what our public API should look like, and what namespaces we want to expose to the user (if any at all). 

See my comment [above](https://github.com/NVIDIA/cccl/pull/3177#discussion_r1887619161).